### PR TITLE
Adds `DynSignatureAlgorithmIdentifier` for PKCS#1 v1.5

### PR DESCRIPTION
**Pull request by baloo**
This is the bits I was missing from https://github.com/RustCrypto/RSA/pull/284

cc @lumag @tarcieri

### CODE REVIEW

File: src/pkcs1v15.rs [hunk 1]
```diff
@@ -732,6 +738,42 @@ where
...
+where
+    D: Digest + AssociatedOid,
+{
+    fn signature_algorithm_identifier(&self) -> AlgorithmIdentifierOwned {
+        AlgorithmIdentifierOwned {
+            oid: match pkcs1v15_oid_for_digest(D::OID) {
+                Some(oid) => oid,
+                None => panic!("no RFC5754 RSA OID defined for Digest"),
+            },
+            parameters: Some(Any::null()),
```
Comment (baloo) [inline src/pkcs1v15.rs]: huuum https://github.com/RustCrypto/formats/pull/969

**Comment by lumag** [comment]
@baloo Being not a fan of `panic!()`, I implemented an alternative solution for the PKCS1v15 keys. Could you please take at https://github.com/RustCrypto/RSA/pull/278/commits/5fb4f893287973b055982846ee08f96b1bc77725

**Comment by baloo** [comment]
I can't make use of that in a code that will be RSA/ecdsa agnostic (like the x509-cert builder) sadly :/
I went with the same design as https://github.com/RustCrypto/signatures/pull/689/files#diff-e83fd22999ab8d7360e0664cabe31e0ed25aca5a41ce1c40be42c5ca3a941cf3R454

**Comment by baloo** [comment]
(I just wired rsa and ecdsa signer in the tests for x509 certs in https://github.com/RustCrypto/formats/pull/764 to give you an idea)

**Comment by lumag** [comment]
> I can't make use of that in a code that will be RSA/ecdsa agnostic (like the x509-cert builder) sadly :/ I went with the same design as https://github.com/RustCrypto/signatures/pull/689/files#diff-e83fd22999ab8d7360e0664cabe31e0ed25aca5a41ce1c40be42c5ca3a941cf3R454

I think, you can, there is a wrapping `SignatureAlgorithmIdentifier` implementation

**Comment by lumag** [comment]
@baloo I think, you can, there is a wrapping SignatureAlgorithmIdentifier implementation
At least it works in my small PKI code path (on par with PSS).

**Comment by baloo** [comment]
Right, that works as expected. Thanks!

**Pull request closed without merging**